src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -670,14 +703,22 @@ - (BOOL)setupCommissioningSessionWithDiscoveredDevice:(MTRCommissionableBrowserR
         return ![MTRDeviceController checkForError:errorCode logMsg:kErrorPairDevice error:error];
     };
 
-    return [self syncRunOnWorkQueueWithBoolReturnValue:block error:error];
+    auto success = [self syncRunOnWorkQueueWithBoolReturnValue:block error:error];
+    if (!success) {
+        MATTER_LOG_METRIC_END(kMetricDarwinFrameworkDeviceCommissioning, errorCode);
+    }
+    return success;
 }
 
 - (BOOL)commissionNodeWithID:(NSNumber *)nodeID
          commissioningParams:(MTRCommissioningParameters *)commissioningParams
                        error:(NSError * __autoreleasing *)error
 {
     auto block = ^BOOL {
+        // Capture metrics on overall status of this stage
+        CHIP_ERROR errorCode = CHIP_NO_ERROR;
+        MATTER_LOG_METRIC_SCOPE(kMetricDarwinFrameworkCommissionNode, errorCode);
```

Comment:
Not sure it makes sense to capture this as begin / end since this only kicks off commissioning?

Reply:
Yup, will change